src/main/java/com/tcg/lwjgllearning/graphics/g3d/mesh/Mesh.java

```diff
@@ -64,8 +64,10 @@ private void createBuffers() {
             positionBuffer = MemoryUtil.memAllocFloat(this.positionArray.length);
             positionBuffer.put(this.positionArray).flip();
 
-            this.positionAttribLocation = this.material.shaderProgram.getAttribLocation("vertPosition");
-            this.positionVboId = glGenBuffers();
+            this.attributeBufferMap = new HashMap<>();
+
+            this.positionAttribLocation = this.shaderProgram.getAttribLocation("vertPosition");
+            this.positionVboId = createAttributeBuffer(Attribute.POSITION);
```

Comment:
Please use `this`. notation.
You can enable IntelliJ to warn when `this.` is not used by enabling the following settings:

File > Settings > [Project Settings] > Inspections > Code Style Issues >
- "Instance method call not qualified with 'this'"
- "Instance fields access not qualified with 'this'"

Reply:
Oops, missed one! I'm using VSCode, but I will find the setting there ^^